packages/lodestar/src/api/impl/validator/duties/proposer.ts

```diff
@@ -0,0 +1,24 @@
+import {IBeaconConfig} from "@chainsafe/eth2.0-config";
+import {IBeaconDb} from "../../../../db/api";
+import {BLSPubkey, Epoch, Slot} from "@chainsafe/eth2.0-types";
+import {computeStartSlotAtEpoch, getBeaconProposerIndex, processSlots} from "@chainsafe/eth2.0-state-transition";
+
+export async function getEpochProposers(
+  config: IBeaconConfig,
+  db: IBeaconDb,
+  epoch: Epoch
+): Promise<Map<Slot, BLSPubkey>> {
+  const state = await db.state.getLatest();
+  //TODO: assert epoch isn't too far ahead
```

Comment:
Is this based on wall clock or based on recent state?

Reply:
It's more about ddos protection, since somebody could send you +20 epoch and your node would stuck as it would try to process huge amount of slot and epoch transitions